toucan_connectors/awsathena/awsathena_connector.py

```diff
@@ -83,20 +84,60 @@ def _add_pagination_to_query(
             return f'SELECT * FROM ({cls._strip_trailing_semicolumn(query)}) LIMIT {limit};'
         return query
 
+    @cached_property
```

Comment:
I'm not sure this is a good idea for long-running processes, maybe we could add a TTL to this cache ?